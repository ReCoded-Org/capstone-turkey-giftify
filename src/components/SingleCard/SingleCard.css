.productCard {
  border-radius: 10px;
  background-color: #90e0ef;
  display: flex;
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font: 1em;
  margin: 10%;
  padding: 3%;
  flex-direction: row;
}

.pruductDetails p {
  background-color: #f8f9fa;
  border-style: ridge;
  border-width: 1px;
  border-color: #006eff;
  border-radius: 5px;
  padding: 1%;
}

.pruductDetails {
  margin-left: 5%;
  width: 60% ;
}

.productCard form {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: center;
}

.submit {
  height: 30px;
  border-radius: 20px;
  margin: 4% 40% 0% 40%;
  background-color: #023e8a;
  color: white;
}

.message {
  padding: 1%;
  width: 100%;
  border-color: #006eff;
  border-radius: 10px;
  background-color: white;
  color: black;
}

.productImage {
  margin-right: 10%;
  border-radius: 5%;
  width: 250px;
}

@media only screen and (min-width: 120px) and (max-width: 531px) {
  .productCard {
    margin: 10%;
    padding: 5%;
    flex-direction: column;
    align-items: center;
    font-size: 1em;
  }

  .submit {
    margin: 10% 30% 10% 30%;
  }

  .productImage {
    margin-right: 10%;
    border-radius: 5%;
    width: 150px;
    margin-bottom: 40px;
  }

  .pruductDetails {
    margin-left: 0%;

  }

}

@media only screen and (min-width: 120px) and (max-width: 767px){

  .productCard {
    margin: 10%;
    padding: 5%;
    flex-direction: column;
    align-items: center;
    font-size: 1em;
  }

  .productImage {
    margin-right: 10%;
    border-radius: 5%;
    width: 200px;
    margin-bottom: 40px;
  }

  .pruductDetails {
    margin-left: 0%;

  }
  
}