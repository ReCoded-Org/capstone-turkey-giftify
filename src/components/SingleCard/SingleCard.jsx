import React, { useState, useEffect } from "react";
import "./SingleCard.css";
import db from "../../firebase";

function SingleCard({ itemsDetails }) {
  const [inputUserMassge, setInputUserMassge] = useState("");
  const [userComments, setUserComments] = useState([]);
  const [userId, setUserId] = useState("2");

  const handleSubmit = (event) => {
    event.preventDefault();
    db.collection("categories")
      .doc(itemsDetails.productType)
      .collection("items")
      .doc(itemsDetails.ItemDocID)
      .collection("requests")
      .add({
        userId: userId,
        readon: inputUserMassge,
      })
      .then(() => {
        setInputUserMassge("");
      })
      .then(() => {
        alert("Your message has been successfully sended");
      });
  };

  useEffect(() => {
    const getData = () => {
      db.collection("categories")
        .doc(itemsDetails.productType)
        .collection("items")
        .onSnapshot(function (querySnapshot) {
          setUserComments(
            querySnapshot.docs.map((doc) => ({
              ...doc.data(),
              userId: doc.data().userId,
              readon: doc.data().readon,
            }))
          );
        });
        
    };
    getData();
  }, [itemsDetails.productType]);

  return (
    <div className="productCard">
      <div>
        <img
          className="productImage"
          src={itemsDetails.productImage}
          alt={itemsDetails.productName}
        />
      </div>
      <div className="pruductDetails">
        <h5>Product Name</h5>
        <p>{itemsDetails.productName}</p>
        <h5>Product Type</h5>
        <p>{itemsDetails.productType}</p>
        <h5>Productâ€™s Short Brief</h5>
        <p>{itemsDetails.ProductDescription}</p>
        <h5>Product Condition</h5>
        <p>{itemsDetails.ItemCondition}</p>
        <form onSubmit={handleSubmit}>
          <label>
            <h5>Why do you need?</h5>
             <textarea
                className="message"
                type="message"
                name="name"
                data-gramm_editor="false"
                value={inputUserMassge}
                onChange ={(e) => setInputUserMassge(e.target.value)}
              />
          </label>
          <br/>
          <input className = "submit" type="submit" value="Submit" />
        </form>
      </div>
    </div>
  );
}

export default SingleCard;
