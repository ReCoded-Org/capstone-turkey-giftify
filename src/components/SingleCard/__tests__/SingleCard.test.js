import React from "react";
import SingleCard from "../SingleCard";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Single Card Page rendered correctly", () => {
  const tree = renderer.create(<SingleCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
