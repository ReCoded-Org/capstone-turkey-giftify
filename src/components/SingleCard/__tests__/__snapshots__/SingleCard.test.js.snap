// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Card Page rendered correctly 1`] = `
<div
  className="productCard"
>
  <div>
    <img
      className="productImage"
    />
  </div>
  <div
    className="pruductDetails"
  >
    <h5>
      Product Name
    </h5>
    <p />
    <h5>
      Product Type
    </h5>
    <p />
    <h5>
      Productâ€™s Short Brief
    </h5>
    <p />
    <h5>
      Product Condition
    </h5>
    <p />
    <form
      onSubmit={[Function]}
    >
      <label>
        <h5>
          Why do you need?
        </h5>
        <textarea
          className="message"
          data-gramm_editor="false"
          name="name"
          onChange={[Function]}
          type="message"
          value=""
        />
      </label>
      <br />
      <input
        className="submit"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
